### DISCUSSION TITLE
Having some issue with channel.onDisconnect()

### DISCUSSION THREAD

**Discussion by HuntingtonJ** [geckosio/geckos.io#149] [Help / Questions] [answered]
Currently working on a multiplayer game (go figure), and for while wasn't having issues with disconnects so not sure if a recent update change things.

Setup:

- 3 node based servers running on amazon ec2 (EDIT: problem does not occur locally)
- 1 server provides a nuxt website
- 1 server hosts the game client
- 1 server manages game server instances through pm2

The basic flow I've implemented is:

1. Server is booted up and waits for a user(=channel) to connect.
2. On the server io.onConnect(), I setup all the basic events, including user.onDisconnect()
3. Server sends a 'ready' event to the connected user
4. User and Server perform a handshake to give the user a unique Id
5. With normal operation, User sends inputs updates and server sends state updates

My issue at the moment is with the user.onDisconnect() event on the server. When I have multiple users connected, and then disconnect one by closing the browser tab, the server io seems to die as evidenced by the connected clients channel.onDisconnect() event firing. The expected behavior is when the server user.onDisconnect() is fired (which it does), it would then send a 'removePlayer' event to the remaining connections (which does not occur, never seen by the other users).

So to summarize, users can connect to the server and play (so server and clients do communicate) but when a user disconnects, the server io instance dies(?) and stop communicating.

Server io
![server_io](https://user-images.githubusercontent.com/22991440/147517299-02f6c33a-18e6-43ac-987f-fce55ca5c1b6.png)

Client boot scene
![image](https://user-images.githubusercontent.com/22991440/147517341-4e5bc9d3-0ed8-46eb-9c8d-0930356c4651.png)

Client game scene (in create function)
![image](https://user-images.githubusercontent.com/22991440/147517369-ec673974-4301-43db-b499-7587b42b9b3b.png)

**Comment by yandeu** [comment, marked as answer]
There was a recent update/fix in onDisconnect().

Can you please open a new issue a provide a small sample project?

I will have a closer look in mid January.

**Comment by Zenchess** [comment]
I have the same issue when I use the express server as https for the game.  Everything works fine but if a player disconnects everyone gets disconnected.  It does not happen when using just a local server without express.

**Reply by yandeu** [reply]
@Zenchess Can you please tell me if the issue goes away if you downgrade geckos to v2.1.4.

**Comment by Zenchess** [comment]
I apologize, I'm pretty sure I had it set up wrong.  It works now, at least with the latest code and I think 2.1.0 as well.
